data-processing/common/commands/fetch_s2_data.py

```diff
@@ -102,10 +105,20 @@ def process(self, item: ArxivId) -> Iterator[S2Metadata]:
                     authors = []
                     for author_data in reference_data["authors"]:
                         authors.append(Author(author_data["authorId"], author_data["name"]))
+
+                    external_ids = reference_data["externalIds"]
+                    if external_ids:
+                        corpus_id = external_ids.get("CorpusId")
+                        arxiv_id = external_ids.get("ArXiv")
+                        doi = external_ids.get("DOI")
+                    else:
+                        corpus_id = arxiv_id = doi = None
+
                     reference = Reference(
                         s2_id=reference_data["paperId"],
-                        arxivId=reference_data["arxivId"],
```

Comment:
Making sure I follow... This would also result in an optional arXiv id and optional doi, because in the old endpoint's response shape, when we don't have these for a reference, the key still exists, and the value is `null`. Is that right?

Reply:
Yup.

With [OLD API]((https://api.semanticscholar.org/v1/paper/arXiv:1512.02595))
-------
We can see references that has a null ArXiv and/or null DOI.  Here's one with both being null:
![Screen Shot 2023-02-06 at 1 44 00 PM](https://user-images.githubusercontent.com/47910008/217094737-cc9b9d8e-ddad-4a30-a5f5-84474ac4bb4f.png)

So the old code was making `Reference` obj like this: 
``` 
reference = Reference(
    s2_id=reference_data["paperId"], <-- actual value
    arxivId=reference_data["arxivId"], <-- None
    doi=reference_data["doi"], <-- None
    ... )
```


With [NEW API](https://api.semanticscholar.org/graph/v1/paper/arXiv:1512.02595?fields=references.authors,references.title,references.externalIds,references.venue,references.year)
-------
That same reference item shows up as:
![Screen Shot 2023-02-06 at 1 50 37 PM](https://user-images.githubusercontent.com/47910008/217095877-311df067-9b76-4105-9b7f-c71cdad5b0f7.png)

So the new code was making `Reference` obj the same way:
```
reference = Reference(
    s2_id=reference_data["paperId"],  <-- actual value
    corpus_id=corpus_id, <-- None
    arxivId=arxiv_id, <-- None
    doi=doi, <-- None
    ...)
```

And for a reference item that has null under `externalIds`, such as this:
![Screen Shot 2023-02-06 at 1 55 07 PM](https://user-images.githubusercontent.com/47910008/217096600-ccb59315-33d3-4c36-8077-a12b90b36bd8.png)
... we'd still be making the `Reference` obj the same way above, except this time s2_id is also None.